rootfs/ingest/python/qserv/contribution.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2014-2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+Helper for ingest contribution management
+
+@author  Fabrice Jammes, IN2P3
+"""
+
+# -------------------------------
+#  Imports of standard modules --
+# -------------------------------
+import logging
+import urllib.parse
+
+# ----------------------------
+# Imports for other modules --
+# ----------------------------
+from .exception import IngestError,  ReplicationControllerError
+from .http import Http
+from .jsonparser import ContributionState, get_contribution_status, raise_error
+from .metadata import LoadBalancedURL
+from .replicationclient import ReplicationClient
+
+# ---------------------------------
+# Local non-exported definitions --
+# ---------------------------------
+_LOG = logging.getLogger(__name__)
+
+# Max attempts to retry ingesting a file on replication service retriable error
+MAX_RETRY_ATTEMPTS = 3
+
+
+def build_contributions(contributions_locked: list,
+                        repl_client: ReplicationClient,
+                        load_balanced_base_url: LoadBalancedURL) -> list:
+    """Build list of contributions to be ingested
+
+    Args:
+        contributions_locked (_type_): _description_
+        repl_client (_type_): _description_
+        load_balanced_url (list): _description_
+
+    Returns:
+        list: contributions to be ingested
+
+    """
+    contributions = []
+    for i, contribution in enumerate(contributions_locked):
+        (database, chunk_id, path, is_overlap, table) = contribution
+
+        (host, port) = repl_client.get_chunk_location(chunk_id, database)
+        contribution = Contribution(host, port,
+                                    chunk_id,
+                                    path,
+                                    table,
+                                    is_overlap,
+                                    load_balanced_base_url)
+        contributions.append(contribution)
+    return contributions
+
+
+class Contribution():
+    """Represent an ingest contribution
+    """
+
+    def __init__(self, worker_host, worker_port, chunk_id, path, table, is_overlap,
```

Comment:
Don't you want to do type annotations for the input parameters?
```
    def __init__(self, worker_host: str, worker_port: int, chunk_id: int, path: str, table: str, is_overlap: bool,
```

Reply:
:+1: